configure.ac

```diff
@@ -575,13 +575,25 @@ AS_IF([test "x$with_gc" = xboehm],
   ]
 )
 
+dnl
+dnl Define kernel version
+dnl
+
+
+gap_kernel_major_version=4
+gap_kernel_minor_version=0
+AC_SUBST([gap_kernel_major_version])
+AC_SUBST([gap_kernel_minor_version])
+
+AC_DEFINE_UNQUOTED([GAP_KERNEL_MAJOR_VERSION], $gap_kernel_major_version, [The major version of the kernel ABI])
+AC_DEFINE_UNQUOTED([GAP_KERNEL_MINOR_VERSION], $gap_kernel_minor_version, [The minor version of the kernel ABI])
```

Comment:
If we do this, we might as well start using this to version the `libgap` shared library, by passing.Not at all saying this should be done in this PR (it's a bit tricky to get semantically right, though not much if one takes the time to read the explanations on what the arguments mean), just wanted to mention it -- we could file an issue for that idea, then somebody (possibly me) can one day look into it.